catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/FeedbackApplication.java

```diff
@@ -0,0 +1,264 @@
+
+    private static final ObjectMapper MAPPER = JsonFactory.create();
+
+    private ConfigurationApplication configurationApplication = null;
+
+    private Template subjectTemplate = null;
+
+    private Template bodyTemplate = null;
+
+    private SmtpClient smtpClient = null;
+
+    private String emailDestination = null;
+
+    @Override
+    public void init() {
+        post("/feedback", APPLICATION_JSON, (req, res) -> {
+            try {
+                FeedbackRequest feedback = parseFeedbackRequest(req.body());
+                feedback.setAuthUsername(getCurrentUser());
+
+                String emailSubject = getEmailSubject(feedback);
+                String emailBody = getEmailBody(feedback);
+                emailBody = emailBody.replaceAll("\\\\n", "\n");
+
+                Session emailSession = smtpClient.createSession();
+                MimeMessage message = new MimeMessage(emailSession);
+                message.addRecipient(Message.RecipientType.TO, new InternetAddress(emailDestination));
+                message.setSubject(emailSubject);
+                BodyPart messageBodyPart = new MimeBodyPart();
+                messageBodyPart.setText(emailBody);
+                Multipart multipart = new MimeMultipart();
+                multipart.addBodyPart(messageBodyPart);
+                message.setContent(multipart);
+                smtpClient.send(message);
+
+                res.body("{}");
+                res.status(200);
+                return res;
+            } catch (Exception e) {
+                LOGGER.debug("Unable to send email", e);
+                throw e;
+            }
+        });
+
+        exception(Exception.class, (e, request, response) -> {
+            response.status(500);
+            response.body("Error submitting feedback");
+            LOGGER.error("Feedback submission failed", e);
+        });
+
+        enableRouteOverview();
+    }
+
+    public void setConfigurationApplication(ConfigurationApplication configurationApplication) {
+        this.configurationApplication = configurationApplication;
+        initTemplates(true);
+    }
+
+    public void setSmtpClient(SmtpClient smtpClient) {
+        this.smtpClient = smtpClient;
+    }
+
+    public String getEmailDestination() {
+        return emailDestination;
+    }
+
+    public void setEmailDestination(String emailDestination) {
+        this.emailDestination = emailDestination;
+    }
+
+    public void refresh(Map<String, Object> configuration) {
+        if (MapUtils.isEmpty(configuration)) {
+            return;
+        }
+
+        SmtpClient smtpClient = (SmtpClient)configuration.get("smtpClient");
+        if (smtpClient != null) {
+            this.smtpClient = smtpClient;
+        }
+
+        this.emailDestination = configurationApplication.getQueryFeedbackEmailDestination();
+        initTemplates(true);
+    }
+
+    private static FeedbackRequest parseFeedbackRequest(String json) {
```

Comment:
There are a lot of casts in this method. Are you sure they are all safe and guaranteed to work?